rules-tests/TypeDeclaration/Rector/ClassMethod/ReturnTypeFromStrictParamRector/Fixture/parent_overridden.php.inc

```diff
@@ -0,0 +1,27 @@
+<?php
+
+namespace Rector\Tests\TypeDeclaration\Rector\ClassMethod\ReturnTypeFromStrictParamRector\Fixture;
+
+use Rector\Tests\TypeDeclaration\Rector\ClassMethod\ReturnTypeFromStrictParamRector\Source\ParentOverrideBase;
+
+class ParentOverridden extends ParentOverrideBase {
+    public function doFoo(ParentOverrideBase $param) {
+        return $param;
+    }
+}
+
+?>
+-----
+<?php
+
+namespace Rector\Tests\TypeDeclaration\Rector\ClassMethod\ReturnTypeFromStrictParamRector\Fixture;
+
+use Rector\Tests\TypeDeclaration\Rector\ClassMethod\ReturnTypeFromStrictParamRector\Source\ParentOverrideBase;
+
+class ParentOverridden extends ParentOverrideBase {
+    public function doFoo(ParentOverrideBase $param): ParentOverrideBase {
```

Comment:
this is allowed, see https://3v4l.org/UiFd6

Reply:
:+1: